### PR TITLE
📝 Component and ComponentManager

### PR DESCRIPTION
**Pull request by je-cook**
## Description

Componet and ComponentManager docs, based off #1567 

## Interface Changes

None

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: documentation/source/base/design_build.dot [hunk 1]
```diff
@@ -26,15 +26,15 @@ digraph {
     subgraph cluster_0 {
         label="Designer";
         tooltip="Designer";
-        href="../base/design_build.html"
+        href="../base/design_build.html#designers"
```
Comment (je-cook) [inline documentation/source/base/design_build.dot:29]: Note that these links now go to the headers of the respective documents. I'm unsure whether we should or just link to the page as there is some overview at the top (same for the rest in this file

File: documentation/source/base/components.rst [hunk 2]
```diff
@@ -1,20 +1,51 @@
...
+
+ComponentManagers
+-----------------
+
+`ComponentManagers` are designed to be created by the Reactor Designer.
+The aim is to make it easier to access logically associated properties of a `Component` that may not be directly connected to the physical object.
+It also can contain helper methods to ease access of specific sections of geometry,
+for instance the separatrix of a plasma.
+
+.. code-block:: pycon
```
Comment (hsaunders1904) [inline documentation/source/base/components.rst:37]: Any reason to prefer `pycon` to just `python` here?

**Review by hsaunders1904 — COMMENTED** [review]
Looks good, a couple of small comments/suggestions

File: documentation/source/base/components.rst (hunk 2 above)
Comment (je-cook) [reply documentation/source/base/components.rst:37]: pycon is for python console output. There is no reason for this to be a console output. I will change it to python.

fyi info about the linting is here https://pygments.org/docs/lexers/

File: documentation/source/base/components.rst [hunk 3]
```diff
@@ -1,20 +1,51 @@
...
+  When used directly it allows other types of component to be grouped together into super-structures.
+  For example, a blanket first wall, breeding zone, manifold, and back supporting structure may be grouped into a blanket system.
+  As these objects are not physical, they do not have a corresponding shape or material composition.
 - :py:class:`bluemira.base.components.PhysicalComponent` defines the physical parts of a
-  reactor, such as blanket layers, vessel shells, or ports. As implementations of
-  :py:class:`bluemira.base.components.PhysicalComponent` correspond to a physical object,
+  reactor, such as blanket layers, vessel shells, or ports.
+  As implementations of :py:class:`bluemira.base.components.PhysicalComponent` correspond to a physical object,
   instances of that class can be defined with a shape and a material.
 - :py:class:`bluemira.base.components.MagneticComponent` defines the magnetic parts of a
```
[the remark this answers is not in the document]
Comment (je-cook) [reply documentation/source/base/components.rst:25]: hmm I cant remember tbh. I will remove it for now and we can have a discussion at some point. I can seef `MagneticComponent` being both confusing and helpful

**Review by hsaunders1904 — APPROVED** [review]

**Pull request merged**